profiler/src/ProfilerEngine/Datadog.Profiler.Native/StackSamplerLoop.cpp

```diff
@@ -510,79 +576,43 @@ void StackSamplerLoop::LogEncounteredStackSnapshotResultStatistics(std::int64_t
     }
 }
 
-uint64_t GetThreadCpuTime(ManagedThreadInfo* pThreadInfo)
-{
-    uint64_t duration = 0;
-
-#ifdef _WINDOWS
-    FILETIME creationTime, exitTime = {}; // not used here
-    FILETIME kernelTime = {};
-    FILETIME userTime = {};
-
-    if (::GetThreadTimes(pThreadInfo->GetOsThreadHandle(), &creationTime, &exitTime, &kernelTime, &userTime))
-    {
-        uint64_t milliseconds = GetTotalMilliseconds(userTime) + GetTotalMilliseconds(kernelTime);
-        return milliseconds;
-    }
-#else
-    // TODO: find the corresponding Linux implementation
-#endif
-
-    return duration;
-}
-
-void StackSamplerLoop::PersistStackSnapshotResults(StackSnapshotResultBuffer const* pSnapshotResult, ManagedThreadInfo* pThreadInfo)
+void StackSamplerLoop::PersistStackSnapshotResults(
+    StackSnapshotResultBuffer const* pSnapshotResult,
+    ManagedThreadInfo* pThreadInfo,
+    PROFILING_TYPE profilingType)
 {
     if (pSnapshotResult == nullptr || pSnapshotResult->GetFramesCount() == 0)
     {
         return;
     }
 
-    // add the WallTime sample to the lipddprof pipeline
-    RawWallTimeSample rawSample;
-    rawSample.Timestamp = pSnapshotResult->GetUnixTimeUtc();
-    rawSample.LocalRootSpanId = pSnapshotResult->GetLocalRootSpanId();
-    rawSample.SpanId = pSnapshotResult->GetSpanId();
-    rawSample.AppDomainId = pSnapshotResult->GetAppDomainId();
-    pSnapshotResult->CopyInstructionPointers(rawSample.Stack);
-    rawSample.ThreadInfo = pThreadInfo;
-    pThreadInfo->AddRef();
-    rawSample.Duration = pSnapshotResult->GetRepresentedDurationNanoseconds();
-    _pWallTimeCollector->Add(std::move(rawSample));
-
-    if (_pConfiguration->IsCpuProfilingEnabled())
+    if (profilingType == PROFILING_TYPE::WallTime)
+    {
+        // add the WallTime sample to the lipddprof pipeline
+        RawWallTimeSample rawSample;
+        rawSample.Timestamp = pSnapshotResult->GetUnixTimeUtc();
+        rawSample.LocalRootSpanId = pSnapshotResult->GetLocalRootSpanId();
+        rawSample.SpanId = pSnapshotResult->GetSpanId();
+        rawSample.AppDomainId = pSnapshotResult->GetAppDomainId();
+        pSnapshotResult->CopyInstructionPointers(rawSample.Stack);
+        rawSample.ThreadInfo = pThreadInfo;
+        pThreadInfo->AddRef();
+        rawSample.Duration = pSnapshotResult->GetRepresentedDurationNanoseconds();
+        _pWallTimeCollector->Add(std::move(rawSample));
+    }
+    else
+    if ((profilingType == PROFILING_TYPE::CpuTime) && (_pConfiguration->IsCpuProfilingEnabled()))
```

Comment:
do we really need to check if it's enabled ?
if not enabled, we won't have Cpu samples, right ?

Reply:
Just in case...
But I'm removing the test
✅